tests/rbd_mirror/test-add-remove-osd-with-mirroring.py

```diff
@@ -0,0 +1,198 @@
+from ceph.parallel import parallel
+from tests.rbd_mirror.rbd_mirror_utils import execute_dynamic, rbd_mirror_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_add_remove_osd_with_mirroring(rbd_mirror, pool_type, **kw):
+    try:
+        mirror1 = rbd_mirror.get("mirror1")
+        mirror2 = rbd_mirror.get("mirror2")
+        config = kw.get("config")
+        pool = config[pool_type]["pool"]
+        image = config[pool_type]["image"]
+        imagespec = pool + "/" + image
+
+        results = {}
+
+        with parallel() as p:
+            p.spawn(
+                execute_dynamic,
+                mirror1,
+                "benchwrite",
+                results,
+                imagespec=imagespec,
+                io="100M",
+            )
+            p.spawn(
+                execute_dynamic,
+                mirror1,
+                "add_remove_osd",
+                results,
+                command="add",
+                osd_nodes=config["osds_to_add"],
+                config=kw.get("config"),
+            )
+
+        if results["benchwrite"] or results["add_remove_osd"]:
+            log.error(f"Results of benchwrite : {results['benchwrite']}")
+            log.error(f"Results of add_remove_osd : {results['add_remove_osd']}")
+            log.error("Adding OSDs failed for primary cluster")
+            return 1
+        try:
+            mirror1.check_data(mirror2, imagespec)
+        except BaseException:
+            log.error(
+                "Data consistency check failed after adding OSDs to primary cluster"
+            )
+            return 1
+
+        with parallel() as p:
+            p.spawn(
+                execute_dynamic,
+                mirror1,
+                "benchwrite",
+                results,
+                imagespec=imagespec,
+                io="100M",
+            )
+            p.spawn(
+                execute_dynamic,
+                mirror1,
+                "add_remove_osd",
+                results,
+                command="rm",
+                osd_nodes=config["osds_to_remove"],
+                config=kw.get("config"),
+            )
+
+        if results["benchwrite"] or results["add_remove_osd"]:
+            log.error(f"Results of benchwrite : {results['benchwrite']}")
+            log.error(f"Results of add_remove_osd : {results['add_remove_osd']}")
+            log.error("Removing OSDs failed for primary cluster")
+            return 1
+
+        try:
+            mirror1.check_data(mirror2, imagespec)
```

Comment:
Can we use "repeat_count" config here, as we are doing same steps two times.

Reply:
we are doing the steps once for primary mirror and once for secondary mirror. not sure if we can use a repeat count thing here. Will try to do it or update here if not possible.

Reply:
ok sure